pkg/providers/kube/client.go

```diff
@@ -15,8 +15,18 @@ import (
 	"github.com/openservicemesh/osm/pkg/service"
 )
 
-// NewProvider implements mesh.EndpointsProvider, which creates a new Kubernetes cluster/compute provider.
-func NewProvider(kubeController k8s.Controller, providerIdent string, cfg configurator.Configurator) (endpoint.Provider, error) {
```

Comment:
Let's just have one function, but have it return the *Client, instead of a provider. Go understands how to handle this.

https://medium.com/@cep21/what-accept-interfaces-return-structs-means-in-go-2fe879e25ee8

Reply:
We can call that function `New`